hadoop-ozone/client/src/test/java/org/apache/hadoop/ozone/client/rpc/RpcClientTest.java

```diff
@@ -0,0 +1,197 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.ozone.client.rpc;
+
+
+import org.apache.hadoop.hdds.protocol.proto.HddsProtos;
+import org.apache.hadoop.ozone.om.helpers.ServiceInfo;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.LinkedList;
+import java.util.List;
+
+import static org.apache.hadoop.ozone.OzoneConfigKeys.OZONE_OM_CLIENT_PROTOCOL_VERSION;
+import static org.apache.hadoop.ozone.client.rpc.RpcClient.validateOmVersion;
+
+/**
+ * Run RPC Client tests.
+ */
+public class RpcClientTest {
+  private enum ValidateOmVersionTestCases {
+    NULL_EXPECTED_NO_OM(
+        null, // Expected version
+        null, // First OM Version
+        null, // Second OM Version
+        true), // Should validation pass
+    NULL_EXPECTED_ONE_OM(
+        null,
+        OZONE_OM_CLIENT_PROTOCOL_VERSION,
+        null,
+        true),
+    NULL_EXPECTED_TWO_OM(
+        null,
+        OZONE_OM_CLIENT_PROTOCOL_VERSION,
+        OZONE_OM_CLIENT_PROTOCOL_VERSION,
+        true),
+    NULL_EXPECTED_TWO_OM_SECOND_MISMATCH(
+        null,
+        OZONE_OM_CLIENT_PROTOCOL_VERSION,
+        "invalid",
+        true),
+    NULL_EXPECTED_TWO_OM_BOTH_MISMATCH(
+        null,
+        "invalid",
+        "invalid",
+        true),
+    NULL_EXPECTED_TWO_OM_FIRST_MISMATCH(
+        null,
+        "invalid",
+        OZONE_OM_CLIENT_PROTOCOL_VERSION,
+        true),
+    EMPTY_EXPECTED_NO_OM(
+        "",
+        null,
+        null,
+        true),
+    EMPTY_EXPECTED_ONE_OM(
+        "",
+        OZONE_OM_CLIENT_PROTOCOL_VERSION,
+        null,
+        true),
+    EMPTY_EXPECTED_TWO_OM(
+        "",
+        OZONE_OM_CLIENT_PROTOCOL_VERSION,
+        OZONE_OM_CLIENT_PROTOCOL_VERSION,
+        true),
+    EMPTY_EXPECTED_TWO_OM_MISMATCH(
+        "",
+        OZONE_OM_CLIENT_PROTOCOL_VERSION,
+        "invalid",
+        true),
+    EMPTY_EXPECTED_TWO_OM_BOTH_MISMATCH(
+        "",
+        "invalid",
+        "invalid",
+        true),
+    VALID_EXPECTED_NO_OM(
+        OZONE_OM_CLIENT_PROTOCOL_VERSION,
+        null,
+        null,
+        false),
+    VALID_EXPECTED_ONE_OM(
+        OZONE_OM_CLIENT_PROTOCOL_VERSION,
```

Comment:
Should nt this be false?
As one OM version is null

Reply:
null == no OM. This means there is a single OM.

Reply:
Yes, missed the != null check when adding to serviceInfo List.